### PR TITLE
SVGLayer image pattern

### PR DESCRIPTION
**Pull request by jonastreub**

### CODE REVIEW

File: framer/SVGLayer.coffee [hunk 1]
```diff
@@ -58,6 +59,25 @@ class exports.SVGLayer extends Layer
...
+		importable: true
+		exportable: true
+		default: null
+		get: -> @_getPropertyValue "imageSize"
+		set: (value) ->
+			if value is null
+				@_setPropertyValue "imageSize", value
+			else
+				return if not _.isFinite(value.width) or not _.isFinite(value.height)
+				@_setPropertyValue "imageSize", {width: value.width, height: value.height}
```
Comment (nvh) [inline framer/SVGLayer.coffee]: Don't you need to update the image pattern after changing the image size?

File: framer/SVG.coffee [hunk 2]
```diff
@@ -31,6 +31,56 @@ class exports.SVG
...
+		transform = ""
+
+		if svgLayer.backgroundSize in ["fill", "fit", "contain", "cover"] and svgLayer.imageSize
+			scaleX = 1
+			scaleY = 1
+			offsetX = 0
+			offsetY = 0
+
+			imageWidth = svgLayer.imageSize.width
+			imageHeight = svgLayer.imageSize.height
```
Comment (nvh) [inline framer/SVG.coffee]: Imagesize can be null

File: framer/SVG.coffee [hunk 3]
```diff
@@ -31,6 +31,56 @@ class exports.SVG
...
+
+			imageRatio = imageWidth / imageHeight
+			realWidth = svgLayer.height * imageRatio
+			realHeight = svgLayer.width / imageRatio
+			scX = realWidth / svgLayer.width
+			scY = realHeight / svgLayer.height
```
Comment (nvh) [inline framer/SVG.coffee]: These can be named bettwr

File: framer/SVG.coffee [hunk 4]
```diff
@@ -31,6 +31,56 @@ class exports.SVG
...
+			transform = """transform="translate(#{offsetX}, #{offsetY}) scale(#{scaleX}, #{scaleY})" """
+
+		if not svgLayer._elementImagePatternSVG
+			svgLayer._elementImagePatternSVG = document.createElementNS("http://www.w3.org/2000/svg", "svg")
+			svgLayer._elementImagePatternSVG.setAttribute("xmlns", "http://www.w3.org/2000/svg")
+			svgLayer._elementImagePatternSVG.setAttribute("width", "100%")
+			svgLayer._elementImagePatternSVG.setAttribute("height", "100%")
+			svgLayer._element.appendChild svgLayer._elementImagePatternSVG
+
+		id = "#{svgLayer.id}-image-pattern"
```
Comment (nvh) [inline framer/SVG.coffee]: Won't this start with a number, and is that allowed in the html spec?

File: framer/SVG.coffee (hunk 2 above)
Comment (jonastreub) [reply framer/SVG.coffee]: I check if imageSize isn't null on line 43

**Pull request merged**